src/models/blogs/blog.service.ts

```diff
@@ -84,4 +84,24 @@ export class BlogService {
 
         await this.blogRepository.remove(blog);
     }
+
+    //Like the post
+    async likeBlog(blogId: string) {
+        const blog = await this.blogRepository.findOneBy({id:blogId})
+
+        if (!blog) throw new Error("Blog not found");
+
+        blog.likes = (blog.likes || 0) + 1;
+        return this.blogRepository.save(blog);
+    }
+
+    //Unlike the blog
+    async unlikeBlog(blogId:string) {
```

Comment:
in order to undo like, you have to keep record of user and liked posts. How do you find out if user likes particular post ?